egs/wsj/s5/steps/cleanup/create_segments_from_ctm_edits.py

```diff
@@ -0,0 +1,724 @@
+    parser.add_argument("eval_in", metavar = "<eval-in>",
+                        help = "Output of 'align-text' post-processed by 'wer_per_utt_details.pl'")
+    parser.add_argument("ctm_in", metavar = "<ctm-in>",
+                        help = "Hypothesized CTM")
+    parser.add_argument("segments_out", metavar = "<segments-out>",
+                        help = "Output sub-segments")
+    parser.add_argument("text_out", metavar = "<text-out>",
+                        help = "Output sub-segment text")
+    args = parser.parse_args()
+
+    return args
+
+def CheckArgs(args):
+    if args.ctm_eval_out is not None:
+        args.ctm_eval_out_handle = open(args.ctm_eval_out, 'w')
+
+    if args.silence_symbol == args.special_symbol:
+        print("WARNING: --silence-symbol and --special-symbol are the same", file = sys.stderr)
+
+    return args
+
+EditType = enum('C', 'S', 'D', 'I', 'EC', 'ES', 'ED', 'EI', 'XS')
+kSilenceEdits = ['D', 'EC', 'ES', 'ED', 'EI']
+
+class CtmEvalProcessor:
+    def __init__(self, args):
+        self.silence_symbol = args.silence_symbol
+        self.special_symbol = args.special_symbol
+        self.oov_symbol = args.oov_symbol
+        self.max_segment_length = args.max_segment_length
+        self.min_segment_length = args.min_segment_length
+        self.max_segment_wer = args.max_segment_wer
+        self.min_silence_length = args.min_silence_length
+        self.min_correct_frames = args.min_correct_frames
+        self.min_wer_for_splitting = args.min_wer_for_splitting
+        self.min_length_for_adding_unk = args.min_length_for_adding_unk
+        self.max_incorrect_words = args.max_incorrect_words
+        self.overlap_length = args.overlap_length
+        self.pad_length = args.pad_length
+        self.frame_shift = args.frame_shift
+        self.length_tolerance = args.length_tolerance
+
+        self.segments_out_handle = open(args.segments_out, 'w')
+        self.text_out_handle = open(args.text_out, 'w')
+
+        self.segments_out = args.segments_out
+        self.text_out = args.text_out
+
+        self.eval_vec = dict()      # key is the utt-id
+        self.ctm = dict()           # key is the utt-id
+        self.ctm_eval = dict()      # key is the utt-id
+
+        # Some statistics for debugging
+        self.num_merged = 0             # Total number of created segments that
+                                        # are merged because they are within
+                                        # length_tolerance distance
+        self.num_padded_frames = 0      # Total number of padded frames
+        self.num_oov_symbols_added = 0  # Total number of oov symbols added
+                                        # in the padded regions to account for
+                                        # partial words
+
+        # Total number of segments processed due to various options
+        self.segments_ignored_on_max_segments_wer = 0
+        self.segments_ignored_on_min_correct_frames = 0
```

Comment:
if you are not using all of this min_correct_frames stuff in the end, best to remove it from everywhere to simplify things- at least in the branch that is to be checked in.